src/main/java/com/microsoft/graph/authentication/TokenCredentialAuthProvider.java

```diff
@@ -0,0 +1,87 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.microsoft.graph.exceptions.AuthenticationException;
+import com.microsoft.graph.exceptions.Error;
+import com.microsoft.graph.httpcore.IHttpRequest;
+import okhttp3.Request;
+import com.microsoft.graph.exceptions.ErrorConstants.*;
+
+import java.util.List;
+
+public class TokenCredentialAuthProvider implements ICoreAuthenticationProvider , IAuthenticationProvider {
```

Comment:
does it really need to implement ICoreAuthenticationProvider.  I'm under the assumption that ICoreAuthenticationProvider is only used when initializing the OkHttpClient which basically does nothing.

Reply:
I believe this was suggestion by Peter to prevent any breaking changes in order to support both current cases.

Reply:
It's used under the cover to authenticate OKHttpRequests in `AuthenticationHandler.java` - https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/dev/src/main/java/com/microsoft/graph/httpcore/AuthenticationHandler.java#L30. 

We should add a work item to ensure we only have `IAuthenticationProvider` and one HTTP request in Java, the way we do in other SDKs. This should be one when we do a complete overhaul of Java since it's a breaking change.